crates/viewer/re_view_spatial/tests/transform_hierarchy.rs

```diff
@@ -0,0 +1,271 @@
+                    // -45 degrees around the y axis.
+                    // Via https://www.andre-gaschler.com/rotationconverter/
+                    re_types::components::RotationQuat(re_types::datatypes::Quaternion::from_xyzw(
+                        [0.0, -0.3826834, 0.0, 0.9238796],
+                    )),
+                ),
+            )
+        });
+
+        path = path.join(&"rotate_mat3x3".into());
+        test_context.log_entity(path.clone(), |builder| {
+            builder.with_archetype(
+                RowId::new(),
+                [(timeline_step, 7)],
+                // -45 degrees around the y axis.
+                // Via https://www.andre-gaschler.com/rotationconverter/
+                &re_types::archetypes::Transform3D::from_mat3x3([
+                    [0.7071069, 0.0000000, -0.7071066],
+                    [0.0000000, 1.0000000, 0.0000000],
+                    [0.7071066, 0.0000000, 0.7071069],
+                ]),
+            )
+        });
+
+        // Add the Rerun asset at the end of the hierarchy.
+        // (We're using a 3D model because it's easier to see the effect of arbitrary transforms here!)
+        {
+            let workspace_dir = std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+            let workspace_dir = workspace_dir
+                .parent()
+                .and_then(|p| p.parent())
+                .and_then(|p| p.parent())
+                .unwrap();
+            let obj_path = workspace_dir.join("tests/assets/rerun.obj");
+
+            path = path.join(&"asset".into());
+            test_context.log_entity(path.clone(), |builder| {
+                builder.with_archetype(
+                    RowId::new(),
+                    [(timeline_step, 0)],
+                    &re_types::archetypes::Asset3D::from_file(&obj_path).unwrap(),
+                )
+            });
+        }
+    }
+
+    let view_id = setup_blueprint(&mut test_context);
+
+    run_view_ui_and_save_snapshot(
+        &mut test_context,
+        timeline_step,
+        view_id,
+        "transform_hierarchy",
+        // TODO(#8924): A lot of pixels won't pass the diff because of anti-aliasing.
+        // By bumping the resolution, we lessen the weight of edges.
+        egui::vec2(300.0, 150.0) * 3.0,
+    );
+}
+
+fn get_test_context() -> TestContext {
+    let mut test_context = TestContext::default();
+
+    // It's important to first register the view class before adding any entities,
+    // otherwise the `VisualizerEntitySubscriber` for our visualizers doesn't exist yet,
+    // and thus will not find anything applicable to the visualizer.
+    test_context.register_view_class::<re_view_spatial::SpatialView3D>();
+
+    // Make sure we can draw stuff in the hover tables.
+    test_context.component_ui_registry = re_component_ui::create_component_ui_registry();
+    // Also register the legacy UIs.
+    re_data_ui::register_component_uis(&mut test_context.component_ui_registry);
```

Comment:
let's drop these from tests that don't draw any component ui
I explored earlier today integrating it into the test context, but that creates very uncomfortable dependencies.